### ISSUE TITLE
remove javadoc from checkstyle

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#3207]
The javadoc checkstyle is getting more in the way than it is helping. 

Example, recent refactor, likely a method was elevated to public, or a line was added, and now needs a javadoc. This potentialy happened off-screen with automatic private->public update.

Now, need to find the one new error in this list:
![screenshot from 2018-03-01 22-04-11](https://user-images.githubusercontent.com/12397753/36885577-a92ea516-1d9c-11e8-8da7-a17ab853c469.png)

This is very tedious, we should probably remove method javadoc requirement until we are ready for it. Interestingly enough the fix for many of these methods is not to javadoc them, but to refactor and reduce scope.

Any really strong objections?

**Comment by DanVanAtta** [comment]
To be clear, I'm proposing we remove method level javadoc requirement; It's the most noisy, not valuable. Simply having methods be protected is not good reason to not javadoc, I don't think we're automating review here; just making life difficult.

**Comment by RoiEXLab** [comment]
No objections.

**Comment by ssoloff** [comment]
No objections.

> Now, need to find the one new error in this list

When I run into this situation, I've found diffing the Checkstyle XML report from the current branch with the report from `master` can usually track down the problem immediately 50% of the time.  Another 30% of the time, it can usually be tracked down in a minute or two.

> Simply having methods be protected is not good reason to not javadoc

True, we should be documenting public and protected members because they are both API.

**Comment by ron-murhammer** [comment]
Fine with me.

**Comment by DanVanAtta** [comment]
Thanks for feedback, I'll submit a PR to remove the method level javadoc requirement.

**Comment by DanVanAtta** [comment]
Looking at the config I noticed the min-line limit is 2. That made me consider maybe we can tweak that. I do put some value in the javadoc requirement, a few times it has reminded me to add a comment. So I experimented with 20 lines and 10 lines instead. There was not much difference in results, seemed like 10 lines would be a more reasonable rule.

A method that long we probably do want to javadoc.
We don't have to worry about methods that we increase from 2 to 3 or 4 lines, all the way up to 10. 
The concern for scope flipping is mostly for the smaller methods.

This PR updated threshold to 10 lines: https://github.com/triplea-game/triplea/pull/3229

I would suggest let's see how this goes, we can revisitin a few days or weeks
